style/rails/README.md

```diff
@@ -14,6 +14,7 @@ Rails
 * Order ActiveRecord associations above ActiveRecord validations.
 * Order controller contents: filters, public methods, private methods.
 * Order i18n translations alphabetically by key name.
+* Split i18n translations into files based on the controller name.
```

Comment:
Since you asked, I'll give this a 👎 .

I don't think this is the right way to group translations. What do you do when the same translation applies to different controllers? Repeat it? What happens if you refactor your controllers; do you now need to repeat the process for your translations? I think the question of how to group translations and how to define controller endpoints are, in general, unrelated.

Reply:
This is in opposition to our recommendations about factories. Is there a reason why one approach would be better for translations vs. factories?

I used to build a separate factory per model and found being able to quickly jump between items in one file was easier to handle.

Reply:
> This is in opposition to our recommendations about factories. Is there a reason why one approach would be better for translations vs. factories?

It's a LOT easier to make sense of a factory definition -- even a long when -- then it is to find a specific fully-qualified YAML key in YAML document with hundreds to thousands of lines. You can do things in YAML files that result in valid YAML but completely obliterate earlier YAML entries, for example. That's a bummer.

It turns out, YAML is really a terrible format for this stuff :)

Reply:
> I don't think this is the right way to group translations. What do you do when the same translation applies to different controllers? Repeat it?

For the most part, yes. It's my experience that "content" translations (as opposed to model field translations, error message translations, etc) tend to be hurt from deduplication. Sometimes the same english text on two pages would be translated to two different words in other languages depending on context, for example. I've found it simpler to not worry about DRYing up translations.

Reply:
>>I don't think this is the right way to group translations. What do you do when the same translation applies to different controllers? Repeat it?

>For the most part, yes. It's my experience that "content" translations (as opposed to model field translations, error message translations, etc) tend to be hurt from deduplication. Sometimes the same english text on two pages would be translated to two different words in other languages depending on context, for example. I've found it simpler to not worry about DRYing up translations.

I agree that it's possible to go too far in DRYing up translations, and that there's a balance to be struck, as with code. I don't think we should make our translations as DRY as possible, necessarily.

My point is that translations are purely presentational, and should therefore not be coupled to controllers, in principle. I was just citing duplication as one possible symptom of what I think the underlying problem is, which is more about separation of concerns.

>It's a LOT easier to make sense of a factory definition -- even a long when -- then it is to find a specific fully-qualified YAML key in YAML document with hundreds to thousands of lines

True, but could we solve that problem better by other means?

For example:

We face a similar problem in managing deeply nested CSS rules. In that case, we typically avoid the problem by flattening it out.

Ex., `.outside { .middle { .inside: ... } }` => `.outside__middle__inside {...}`.

If we did something analogous for translations, it might be easier to search for than either deeply nested YAML files, or YAML files based on controllers.

Anyway, whether we merge this or not, it's not the end of the world. I'm just personally not sure enough of this approach to support making it our "official" policy, and wanted to voice some reservations.